scripts/generate/templates/SafeCast.js

```diff
@@ -91,7 +121,9 @@ const toUintDownCast = length => `\
  * _Available since v${version('toUint(uint)', length)}._
  */
 function toUint${length}(uint256 value) internal pure returns (uint${length}) {
-    require(value <= type(uint${length}).max, "SafeCast: value doesn't fit in ${length} bits");
+    if (value > type(uint${length}).max) {
+      revert SafeCastOverflowedUintDowncast(${Math.min(length, 255)}, value);
```

Comment:
if length is > 255, maybe we should throw an error (during the code generation) rather then doing "Math.min"

Reply:
The reason this is added is because the bit sizes go from `8` to `256`, where `uint8` goes from `0` to `255`, so in order to use an `uint8` without overflowing for `256` to represent a bit size I discussed 2 alternatives with @frangio:

- Representing bit sizes with `n - 1`, (eg. `7, 15, 23, ..., 255`)
- Representing bit sizes with their respective number (eg. `8, 16, 24, ...`) except for `256` which is represented by `255`

Does this make sense to you?

Reply:
My point is that this error will never get trigered with "length" 256:
- by definition a uint256 fits into a uin256
- `value > type(uint256).max` will never trigger

Basically, just trying to instantiate the downcast templates with length 256 is a mistake. IMO there is no point in finding a way to represent 256... because 256 should just not be.

Reply:
Ok, I think the initial comment was not clear with the intention since I thought it was suggesting a Javascript `throw` to avoid Solidity not compiling.

Based on your explanation, I think that's not only a problem with the custom errors or the `Math.min`, but with the error suite as a whole, and it's currently handled in the defaults of the `version` switch statements.

What I'll do is remove the `NOTE`s and the `Math.min` from both `Downcast` errors.

Reply:
1. The initial suggestion _was_ to throw in JavaScript during code generation.
2. It _is_ necessary to find a way to represent 256, for the errors when converting between signed and unsigned integers.

Reply:
> The initial suggestion was to throw in JavaScript during code generation.

Yes, and that's already implemented in the [`version` function](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/cb4594f7b94b9fa411f724daebeb7d0587458f0e/scripts/generate/templates/SafeCast.js#L9), so I don't see why add it only for the error length check

> It is necessary to find a way to represent 256, for the errors when converting between signed and unsigned integers.

We have a note `NOTE: The \`bits\` argument is \`min(bits, 255)\`.` that is kept for such cases.

Reply:
> It _is_ necessary to find a way to represent 256, for the errors when converting between signed and unsigned integers.

- The Downcast conversion don't need 256
- The uint <> int casting only exist for uint256 <> int256. Smaller size are not supported.

Reply:
My comment was for visibility, yes, but I had not realized that uint <> int only exists for 256 bits! In this case it really seems like we don't need the `bits` in that error. I would remove it, yes.

Reply:
I completely missed this. Solving since the suggestion of removing the bitsize in casting between types